database/schema.go

```diff
@@ -6,29 +6,35 @@ import (
 	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
-// schema defines a mapping between a collection name and the indexes that
+// schema returns a mapping between a collection name and the indexes that
 // must exist for that collection.
-var schema = map[string][]mongo.IndexModel{
-	collSkylinks: {
-		{
-			Keys:    bson.D{{"skylink", 1}},
-			Options: options.Index().SetName("skylink").SetUnique(true),
+//
+// We return a map literal instead of using a global variable because the global
```

Comment:
Right, that's what I thought as well. Then I was puzzling over data races for a while and even pulled Chris to help me because I couldn't figure out how this could cause a data race (no code ever changes this in any way). And yet I was getting a data race on this line:
```go
names, err := iv.CreateMany(ctx, models)
```
If you're curious and want to play with it - go for it. I suspect there might be some valuable lesson at the bottom of that hole but I want to get this in prod ASAP, so we can stop losing files.

Reply:
hm well I'm not feeling much for going down that rabbit hole but most likely it's just `CreateMany` doing a set on of the models, if you look at the implementation it does `model.Options.SetName(name)` on line 209 so probably simply doing `names, err := iv.CreateMany(ctx, append([]mongo.IndexModel{}, models...))` gets around the race

Reply:
(obviously fine to resolve)